client/src/app/components/FilterToolbar/MultiselectFilterControl.tsx

```diff
@@ -42,37 +42,43 @@ export const MultiselectFilterControl = <TItem,>({
   IMultiselectFilterControlProps<TItem>
 >): JSX.Element | null => {
   const [isFilterDropdownOpen, setIsFilterDropdownOpen] = React.useState(false);
+  const [inputValue, setInputValue] = React.useState<string>("");
+  const textInputRef = React.useRef<HTMLInputElement>();
 
-  const [selectOptions, setSelectOptions] = React.useState<
-    FilterSelectOptionProps[]
-  >(Array.isArray(category.selectOptions) ? category.selectOptions : []);
-
-  React.useEffect(() => {
-    setSelectOptions(
-      Array.isArray(category.selectOptions) ? category.selectOptions : []
-    );
-  }, [category.selectOptions]);
-
-  const hasGroupings = !Array.isArray(selectOptions);
-
-  const flatOptions: FilterSelectOptionProps[] = !hasGroupings
-    ? selectOptions
-    : (Object.values(selectOptions).flatMap(
-        (i) => i
-      ) as FilterSelectOptionProps[]);
-
-  const getOptionFromOptionValue = (optionValue: string) =>
-    flatOptions.find(({ value }) => value === optionValue);
+  const idPrefix = `filter-control-${category.categoryKey}`;
+  const withPrefix = (id: string) => `${idPrefix}-${id}`;
+  const defaultGroup = category.title;
 
-  const [focusedItemIndex, setFocusedItemIndex] = React.useState<number | null>(
-    null
+  const filteredOptions = category.selectOptions?.filter(({ label, value }) =>
+    String(label ?? value ?? "")
+      .toLowerCase()
+      .includes(inputValue?.trim().toLowerCase() ?? "")
```

Comment:
This will only filter by the option's "label" and does not include the option's "groupLabel".

So trying to filter the select options by a tag category doesn't work.

Reply:
good catch! previously  `label` == `value` in this scenario (and `value` contains both tag category and tag name)